openpocket/alpha_spheres/alpha_spheres.py

```diff
@@ -126,43 +126,44 @@ def remove_alpha_spheres(self, indices):
 
         """
         if indices is not None:
-        #    if isinstance(indices, (list,np.ndarray,dtype:ints)): 
-            if isinstance(indices, (list,np.ndarray)):
+            if isinstance(indices, (list,np.ndarray,tuple)):
                 pass
             else:
-                raise ValueError("The argument points needs to be a List, tuple or numpy.ndarray object with the integer numbers corresponding to the alpha-sphere")
-
-            self.indices = indices  # a way to store the input indices
-            alpha_sphere_index=[]
-            for ia,alpsp in enumerate(self.points_of_alpha_sphere):
-                alpha_sphere_index.append(ia) 
-            for ix in self.indices:
-#                if ix in alpha_sphere_index:
-                alpha_sphere_index.remove(ix)
-               # remaining=[]
-                self.remove_alpha_spheres=[]
-                for ai in alpha_sphere_index:
-                    self.remove_alpha_spheres.append(self.points_of_alpha_sphere[ai])
-               #     self.remove_alpha_spheres = self.points_of_alpha_sphere[ai]
-               # else:
-               #     raise IndexError("has been using a non existent index")
-
+                raise ValueError("The input indices needs to be a List, tuple or numpy.ndarray object with the integer numbers corresponding to the alpha-sphere")
+            for ix in indices:
+                point_index.remove(ix)
+            
+            remaining_centers=[]
+            remainig_points_of_alpha_sphere=[]
+            remainig_points=[]
+            for ri in point_index:
+                remaining_centers.append(self.centers[ri])
+                remainig_points_of_alpha_sphere.append(self.points_of_alpha_sphere[ri])
+                remaining_points.append(self.points[ri]) 
+            self.centers=remaining_centers
+            self.n_alpha_spheres = self.centers.shape[0]
+            self.remove_alpha_spheres=remainig_points_of_alpha_sphere
 
+            self.radii = []
+            for ii in range(self.n_alpha_spheres):
+                radius = euclidean(self.centers[ii], remaining_points[self.remove_alpha_spheres[ii][0]])
+                self.radii.append(radius)
```

Comment:
This is ok, but it is expensive. Why don't you use the same strategy with radii than with `self.centers` or `self.points`?

Reply:
I don't know how to do it, because there is no index for the radii method, and the input index is not the same length as the radius index.